### PR TITLE
[SPARK-28935][SQL][DOCS] Document SQL metrics for Details for Query Plan

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch adds the description of common SQL metrics in web ui document.

### Why are the changes needed?

The current web ui document describes query plan but does not describe the meaning SQL metrics. For end users, they might not understand the meaning of the metrics.

### Does this PR introduce any user-facing change?

No. This is just documentation change.

### How was this patch tested?

Built the docs locally.

image

### CODE REVIEW

**Comment by viirya** [comment]
cc @gatorsmile @dilipbiswal

File: docs/web-ui.md [hunk 1]
```diff
@@ -363,6 +363,42 @@ number of written shuffle records, total data size, etc.
 Clicking the 'Details' link on the bottom displays the logical plans and the physical plan, which
 illustrate how Spark parses, analyzes, optimizes and performs the query.
 
+### SQL metrics
+
+The metrics of SQL operators show in the block of operators. The SQL metrics can be useful when
+we want to dive into the execution details of each operator, for example, how many rows are output
+after a Filter operator. The related metrics are different for each type of operator, for example
+Exchange has the metrics called "shuffle bytes writte total" which shows the number of bytes written
```
Comment (dilipbiswal) [inline docs/web-ui.md]: Nit: written ?

File: docs/web-ui.md [hunk 2]
```diff
@@ -363,6 +363,42 @@ number of written shuffle records, total data size, etc.
...
+<tr><td> <code>data size</code> </td><td> the size of broadcasted/shuffled/collected data of the operator </td><td> BroadcastExchange, ShuffleExchange, Subquery </td></tr>
+<tr><td> <code>time to collect</code> </td><td> the time spent to collect data </td><td> BroadcastExchange, Subquery </td></tr>
+<tr><td> <code>scan time</code> </td><td> the time spent to scan data </td><td> ColumnarBatchScan, FileSourceScan </td></tr>
+<tr><td> <code>metadata time</code> </td><td> the time spent on getting metadata like number of partitions, number of files </td><td> FileSourceScan </td></tr>
+<tr><td> <code>shuffle bytes written</code> </td><td> number of bytes written </td><td> CollectLimit, TakeOrderedAndProject, ShuffleExchange </td></tr>
+<tr><td> <code>shuffle records written</code> </td><td> number of records written </td><td> CollectLimit, TakeOrderedAndProject, ShuffleExchange </td></tr>
+<tr><td> <code>shuffle write time</code> </td><td> the time on shuffle writing </td><td> CollectLimit, TakeOrderedAndProject, ShuffleExchange </td></tr>
```
Comment (dilipbiswal) [inline docs/web-ui.md]: Nit: the time spent on writing shuffle data ?

File: docs/web-ui.md [hunk 3]
```diff
@@ -363,6 +363,42 @@ number of written shuffle records, total data size, etc.
 Clicking the 'Details' link on the bottom displays the logical plans and the physical plan, which
 illustrate how Spark parses, analyzes, optimizes and performs the query.
 
+### SQL metrics
+
+The metrics of SQL operators show in the block of operators. The SQL metrics can be useful when
```
Comment (dilipbiswal) [inline docs/web-ui.md]: @viirya a little confused on "show in the block of operators" ? Is there a way to reword this ?

**Comment by dilipbiswal** [comment]
@viirya Looks good to me. I have some minor comments :-)

**Comment by gatorsmile** [comment]
LGTM after a few minor updates!

**Comment by viirya** [comment]
thanks for updating! @gatorsmile

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**